File: backend/src2/SharedModels/DTOs/InputGeneratorDto.cs
using System.Collections.Generic;

namespace SharedModels.DTOs
{
    public class InputGeneratorDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string CreatedByUser { get; set; }
        public List<GeneratorCallArgumentDto> CallArguments { get; set; }
        public CodeSnippetDto CodeSnippet { get; set; }        
    }
}

File: frontend/src/app/shared/code/InputGeneratorDto.ts
import { CodeSnippet } from '../CodeSnippet'
import { Type } from 'class-transformer'
import { GeneratorCallArgumentDto } from './GeneratorCallArgumentDto'

export class InputGeneratorDto {

  @Type(() => Number)
  id: number = undefined

  name: string

  @Type(() => String)
  description: string = ''

  @Type(() => CodeSnippet)
  codeSnippet: CodeSnippet

  @Type(() => GeneratorCallArgumentDto)
  callArguments: GeneratorCallArgumentDto[]


  constructor(codeSnippet: CodeSnippet = null) {
    this.codeSnippet = codeSnippet
  }

  public static EMPTY(): InputGeneratorDto {
    const snippet = CodeSnippet.EMPTY()
    return new InputGeneratorDto(snippet)
  }
}
